django/db/models/base.py

```diff
@@ -1087,7 +1087,8 @@ def _perform_unique_checks(self, unique_checks):
             for field_name in unique_check:
                 f = self._meta.get_field(field_name)
                 lookup_value = getattr(self, f.attname)
-                if lookup_value is None:
+                if (lookup_value is None or
+                        lookup_value == '' and connection.features.interprets_empty_strings_as_nulls):
```

Comment:
It's not clear to me that this handles multi-db correctly. This line always checks the feature `interprets_empty_strings_as_nulls` on the default database, which may not be the same database we are actually saving to. Am I missing something?